plasmapy/diagnostics/langmuir.py

```diff
@@ -1,153 +1,1073 @@
+                                   0.0, None))
+
+    ion_characteristic = Characteristic(probe_characteristic.bias, ion_current)
+
+    if(visualize):
+        with quantity_support():
+            plt.figure()
+            plt.scatter(probe_characteristic.bias,
+                        probe_characteristic.current.to(u.mA), marker='.',
+                        c='k')
+            plt.plot(probe_characteristic.bias,
+                     ion_characteristic.current.to(u.mA), c='y')
+
+    return ion_characteristic
+
+
+@utils.check_quantity({'probe_area': {'units': u.m**2,
+                                      'can_be_negative': False,
+                                      'can_be_complex': False,
+                                      'can_be_inf': False,
+                                      'can_be_nan': False}})
+def get_EEDF(probe_characteristic, probe_area, n_i, visualize=False):
+    r"""Implement the Druyvesteyn method of obtaining the normalized
+    Electron Energy Distribution Function (EEDF).
+
+    Parameters
+    ----------
+    probe_characteristic : Characteristic
+        The swept probe characteristic that is to be analyzed.
+
+    probe_area : Quantity
+        The area of the probe exposed to plasma in units convertible to m^2.
+
+    n_i : Quantity
+        The ion density in units covertible to m^-3.
+
+    visualize : bool, optional
+        If True a plot of the extracted electron current is shown. Default is
+        False.
+
+    Returns
+    -------
+    energy : Quantity, ndarray
+        Array of potentials in V.
+
+    probability : float, ndarray
+        Array of floats corresponding to the potentials representing the EEDF
+        in normalized probabilities.
+
+    Notes
+    -----
+    The Druyvesteyn method requires the second derivative of the probe I-V
+    characteristic, which inherently amplifies noise and measurement errors.
+    Therefore it is advisable to smooth the I-V prior to the use of this
+    function.
+
+    """
+
+    probe_characteristic.check_validity()
+
+    probe_characteristic.sort()
+
+    characteristic_unique = probe_characteristic.get_unique()
+
+    V_F = get_floating_potential(characteristic_unique)
+
+    V_P = get_plasma_potential(characteristic_unique)
+
+    # Obtain the correct EEDF energy range from the probe characteristic.
+    _filter = (characteristic_unique.bias > V_F) & \
+        (characteristic_unique.bias < V_P)
+    energy = const.e * (V_P - characteristic_unique.bias[_filter])
+    energy = energy.to(u.eV)
+
+    # Below follows a very crude method of obtaining the second derivative of
+    # the U-V curve. Any suggestions?
+    dIdV = np.diff(characteristic_unique.current[_filter]) / \
+        np.diff(characteristic_unique.bias[_filter])
+    dIdV = np.append(dIdV, dIdV[-1])
+    dIdV2 = np.diff(dIdV) / np.diff(characteristic_unique.bias[_filter])
+    dIdV2 = np.append(dIdV2, dIdV2[-1])
```

Comment:
Two applications of [numpy.gradient](https://docs.scipy.org/doc/numpy/reference/generated/numpy.gradient.html) would let you get rid of the appends... I guess you could also, for sufficiently smooth and neat functions, use [scipy.fftpack.diff](https://docs.scipy.org/doc/scipy/reference/generated/scipy.fftpack.diff.html), but that uses the FFT and may glitch out. I guess both might.